core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -442,6 +450,41 @@ private DBWriter() {
 
     }
 
+    public static Future<?> addFavoriteItem(final FeedItem item) {
+        return dbExec.submit(() -> {
+            final PodDBAdapter adapter = PodDBAdapter.getInstance().open();
+            adapter.addFavoriteItem(item);
+            adapter.close();
+            item.addTag(FeedItem.TAG_FAVORITE);
+            EventBus.getDefault().post(FavoritesEvent.added(item));
+        });
+    }
+
+    public static Future<?> addFavoriteItemById(final long itemId) {
+        return dbExec.submit(() -> {
+            final FeedItem item = DBReader.getFeedItem(itemId);
+            if (item == null) {
+                Log.d(TAG, "Can't find item for itemId " + itemId);
```

Comment:
I thought about that, but then there'd be nested dbExec.submit() calls happening. Wasn't sure how big a deal that would be.